src/bors/handlers/review.rs

```diff
@@ -0,0 +1,261 @@
+use crate::PgDbClient;
+
+/// Approve a pull request.
+/// A pull request can only be approved by a user of sufficient authority.
+pub(super) async fn command_approve(
+    repo_state: Arc<RepositoryState>,
+    db: Arc<PgDbClient>,
+    pr: &PullRequest,
+    author: &GithubUser,
+    approver: &Approver,
+) -> anyhow::Result<()> {
+    tracing::info!("Approving PR {}", pr.number);
+    if insufficient_approve_permission(repo_state.clone(), author) {
+        deny_approve_request(&repo_state, pr, author).await?;
+        return Ok(());
+    };
+    let approver = match approver {
+        Approver::Myself => author.username.clone(),
+        Approver::Specified(approver) => approver.clone(),
+    };
+    tracing::info!("Approving PR {} by {}", pr.number, approver);
+    db.approve(repo_state.repository(), pr.number, approver.as_str())
+        .await?;
+    tracing::info!("Approved PR {}", pr.number);
+    handle_label_trigger(&repo_state, pr.number, LabelTrigger::Approved).await?;
+    tracing::info!("Label trigger handled for PR {}", pr.number);
+    notify_of_approval(&repo_state, pr, approver.as_str()).await
+}
+
+/// Unapprove a pull request.
+/// Pull request's author can also unapprove the pull request.
+pub(super) async fn command_unapprove(
+    repo_state: Arc<RepositoryState>,
+    db: Arc<PgDbClient>,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::info!("Unapproving PR {}", pr.number);
+    if insufficient_unapprove_permission(repo_state.clone(), pr, author) {
+        deny_unapprove_request(&repo_state, pr, author).await?;
+        return Ok(());
+    };
+    db.unapprove(repo_state.repository(), pr.number).await?;
+    handle_label_trigger(&repo_state, pr.number, LabelTrigger::Unapproved).await?;
+    notify_of_unapproval(&repo_state, pr).await
+}
+
+fn insufficient_approve_permission(repo: Arc<RepositoryState>, author: &GithubUser) -> bool {
+    !repo
+        .permissions
+        .load()
+        .has_permission(author.id, PermissionType::Review)
+}
+
+async fn deny_approve_request(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    author: &GithubUser,
+) -> anyhow::Result<()> {
+    tracing::warn!(
+        "Permission denied for approve command by {}",
+        author.username
+    );
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "@{}: :key: Insufficient privileges: not in review users",
+                author.username
+            )),
+        )
+        .await
+}
+
+async fn notify_of_approval(
+    repo: &RepositoryState,
+    pr: &PullRequest,
+    approver: &str,
+) -> anyhow::Result<()> {
+    repo.client
+        .post_comment(
+            pr.number,
+            Comment::new(format!(
+                "Commit {} has been approved by `{}`",
+                pr.head.sha, approver
+            )),
+        )
+        .await
+}
+
+fn insufficient_unapprove_permission(
```

Comment:
I find it easier to follow if the logic is positive, rather than negative. Could you please change this into something like `has_[un]approve_permissions`?